prolog.cl

```diff
@@ -841,6 +842,747 @@
+         4
+        '("True")
+        
+         '("False. Neither are allowed"
+           "False. Only synchronous are allowed."
+           "False. Only asynchronous are allowed."
+           )
+)
+
+  (multi "What is a characteristic of the operations of streams in Snowflake?"
+         4
+        '("When a stream is used to update a target table, the offset is advanced to the current time.")
+         '("Whenever a stream is queried, the offset is automatically advanced."
+           "Querying a stream returns all change records and table rows from the current offset to the current time."
+           "Each committed and uncommitted transaction on the source table automatically puts a change record in the stream."
+           )
+)
+
+  (multi "A Data Engineer wants to centralize grant management to maximize security. A user needs OWNERSHIP on a table in a new schema. However, this user should not have the ability to make grant decisions.
+What is the correct way to do this?"
+         4
+        '("Add the WITH MANAGED ACCESS parameter on the schema.")
+        
+         '("Grant OWNERSHIP to the user on the table."
+           "Revoke grant decisions from the user on the table."
+           "Revoke grant decisions from the user on the schema."
+           )
+)
+
+  (multi "A Data Engineer is implementing a near real-time ingestion pipeline to load data into Snowflake using the Snowflake Kafka connector. There will be three Kafka topics created.
+Which Snowflake objects are created automatically when the Kafka connector starts?"
+         4
+        '("Tables"
+          "Pipes"
+          "Internal stages")
+        
+         '("Tasks"
+           "External stages"
+           "Materialized views"
+           )
+)
+
+  (multi "What kind of Snowflake integration is required when defining an external function in Snowflake?"
+         4
+        '(API integration"")
+        
+         '("HTTP integration"
+           "Notification integration"
+           "Security integration"
+           )
+)
+
+  (multi "What is the purpose of the BUILD_STAGE_FILE_URL function in Snowflake?"
+         4
+        '("It generates a permanent URL for accessing files in a stage.")
+        
+         '("It generates an encrypted URL for accessing a file in a stage."
+           "It generates a staged URL for accessing a file in a stage."
+           "It generates a temporary URL for accessing a file in a stage."
+           )
+)
+
+  (multi "A Data Engineer needs to load JSON output from some software into Snowflake using Snowpipe.
+Which recommendation(s) apply to this scenario?"
+         4
+        '("Ensure that data files are 100-250 MB (or larger) in size, compressed."
+         "Extract semi-structured data elements containing null values into relational columns before loading."
+         "Verify each value of each unique element stores a single native data type (string or number).")
```

Comment:
Where's the documentation source for this one?

Reply:
This question was asked verbatim on the actual exam. 

That said, looking for documentation, I found:
https://docs.snowflake.com/en/user-guide/data-load-considerations-prepare#general-file-sizing-recommendations
https://docs.snowflake.com/en/user-guide/semistructured-considerations#how-extraction-impacts-queries